### PR TITLE
`data_rename()`: "Error: Following variable(s) were not found" in version '0.13.0.17'

### LINKED ISSUE #571: `data_rename()`: "Error: Following variable(s) were not found" in version '0.13.0.17'
Yo, trying to integrate the devel version of `datawizard '0.13.0.17'` I'm facing a breaking change in `data_rename()` (more info in easystats/report#470). Just wanted to check with you if that's something I should change in `report` or `datawizard`. My attempt at a minimal reprex:

### `datawizard` '0.13.0.17'

``` r
packageVersion("datawizard")
#> [1] '0.13.0.17'

result <- list(
  setosa.a = structure(data.frame(1)), 
  versicolor.a = structure(data.frame(2)))

new_names <- c("setosa, a", "versicolor, a")

names(result)
#> [1] "setosa.a"     "versicolor.a"
new_names
#> [1] "setosa, a"     "versicolor, a"

datawizard::data_rename(result, select = names(result), replacement = new_names)
#> Error: Following variable(s) were not found: setosa.a, versicolor.a
#>   Possibly misspelled?
```

<sup>Created on 2024-12-12 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

### `datawizard` '0.13.0'

``` r
packageVersion("datawizard")
#> [1] '0.13.0'

result <- list(
  setosa.a = structure(data.frame(1)), 
  versicolor.a = structure(data.frame(2)))

new_names <- c("setosa, a", "versicolor, a")

names(result)
#> [1] "setosa.a"     "versicolor.a"
new_names
#> [1] "setosa, a"     "versicolor, a"

datawizard::data_rename(result, select = names(result), replacement = new_names)
#> $`setosa, a`
#>   X1
#> 1  1
#> 
#> $`versicolor, a`
#>   X2
#> 1  2
```

<sup>Created on 2024-12-12 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

The reason seems to be that `data_rename()` searches for all the variables mentioned in select for each data frame of the list, but each name only appears once in their respective data frame. Therefore, some variables are bound to not be found. So, whereas `data_rename()` used to work on lists, now it only works on dataframes? This seems to break `report::report_sample` tests.

### PR DESCRIPTION
**Pull request by strengejacke**
Fixes #571

### CODE REVIEW

**Comment by strengejacke** [comment]
I think this can be merged, but @etiennebacher, do you know why the docs for `rescale()` inherit the `@details` section from `data_rename()`?

File: R/data_rename.R [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #' @title Rename columns and variable names
+#' @name data_rename
```
Comment (etiennebacher) [inline R/data_rename.R:2]: This is unnecessary since the function is already named `data_rename`

File: tests/testthat/test-data_rename.R [hunk 2]
```diff
@@ -221,3 +221,14 @@ test_that("Argument `pattern` is deprecated", {
     fixed = TRUE
   )
 })
+
+test_that("works with lists", {
+  result <- list(
+    setosa.a = structure(data.frame(1)),
+    versicolor.a = structure(data.frame(2))
+  )
```
Comment (etiennebacher) [inline tests/testthat/test-data_rename.R:229]:
```suggestion
  result <- list(x = 1, y = 2)
```

**Review by etiennebacher — APPROVED** [review]

**Comment by strengejacke** [comment]
> > do you know why the docs for rescale() inherit the `@details` section from data_rename()?
> 
> No idea why, can you remove it?

I tried to, bust failed - however, just realized there is a `@inherit data_rename` tag, which is probably only accidentally there. Removing it fixes the issue.

File: R/data_rename.R (hunk 1 above)
Comment (strengejacke) [reply R/data_rename.R:2]: I know, but I prefer having a consistent structure of the header part of an R file, including explicit tags (else, it depends on the order, which _may_ change).

**Pull request merged**